src/linker/Linker.Dataflow/ControlFlowGraph.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using ILLink.Shared.DataFlow;
+using Mono.Cecil.Cil;
+using System.Diagnostics.CodeAnalysis;
+
+using Predecessor = ILLink.Shared.DataFlow.IControlFlowGraph<
+	Mono.Linker.Dataflow.BasicBlock, 
+	Mono.Linker.Dataflow.Region
+>.Predecessor;
+
+namespace Mono.Linker.Dataflow
+{
+	public record struct Region : IRegion<Region>
+	{
+		public RegionKind Kind => throw new NotImplementedException ();
+	}
+
+	public struct BasicBlock : IEquatable<BasicBlock>
+	{
+		public List<Instruction> Instructions;
+
+		public BasicBlock ()
+		{
+			Instructions = new List<Instruction> ();
+		}
+
+		public bool Equals (BasicBlock other)
+		{
+			return Instructions == other.Instructions;
```

Comment:
This is reference equality, not content equality. Not sure what we need here, but it does feel wrong.
Meaning:
```C#
        List<int> a = new List<int>() { 1, 2 };
        List<int> b = new List<int>() { 1, 2 };
                                       
        Console.WriteLine(a == a);  // true
        Console.WriteLine(b == a);  // false
```